src/jit/codegenarm64.cpp

```diff
@@ -3564,6 +3584,11 @@ void CodeGen::genCodeForStoreInd(GenTreeStoreInd* tree)
             dataReg = data->gtRegNum;
         }
 
+        if (tree->gtFlags & GTF_IND_VOLATILE)
+        {
+            instGen_MemoryBarrier();
+        }
```

Comment:
A comment stating that we issue a full memory barrier before the store operation